tools/find_deprecated.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""List deprecated decorators."""
+from __future__ import annotations
+from typing import cast, Optional
+from pathlib import Path
+from collections import OrderedDict, defaultdict
+import ast
+from datetime import datetime
+import sys
+import requests
+
+
+def short_path(path: Path) -> Optional[Path]:
+    """shorten the full path, when possible"""
+    original_path = path
+    while path is not None:
+        if Path.cwd().is_relative_to(path):
+            return original_path.relative_to(path)
+        path = path.parent if path != path.parent else None
+    return original_path
+
+
+class Deprecation:
```

Comment:
It looks like this only has one subclass, DeprecationCall? You can simplify this by merging the two. No need to overly generalize.

Reply:
`DeprecationDecorator` is also a subclass. It helped me to code it.